### PR TITLE
feat/quasar framework

### PR DESCRIPTION
**Pull request by fmarahre**
- added quasar framework to the project
- components redesigned with quasar

### CODE REVIEW

File: vue/src/views/DashboardView.vue [hunk 1]
```diff
@@ -55,19 +65,34 @@ import { weatherDataStore } from "../stores/weatherDataStore";
...
+// Sensor - Map interaction
+function mapSensorEnter(sensorId: number) {
+  sensorCompRef.value?.setRowActive(sensorId);
+}
+
+function mapSensorLeave(sensorId: number) {
+  sensorCompRef.value?.setRowInactive(sensorId);
+}
+
+function tableSensorEnter(sensorId: number) {
```
Comment (lucaslootalot) [inline vue/src/views/DashboardView.vue:91]: return type added
```suggestion
function tableSensorEnter(sensorId: number): void {
```

File: vue/src/views/DashboardView.vue [hunk 2]
```diff
@@ -55,19 +65,34 @@ import { weatherDataStore } from "../stores/weatherDataStore";
...
+  gardenMapRef.value?.setMarkerActive(sensorId);
+}
+
+function tableSensorLeave(sensorId: number) {
```
Comment (lucaslootalot) [inline vue/src/views/DashboardView.vue:95]: return type added
```suggestion
function tableSensorLeave(sensorId: number): void {
```

**Review by lucaslootalot — CHANGES_REQUESTED** [review]

File: vue/src/components/CropsTable.vue [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+import { ref } from "vue";
+import type { QTableProps } from "quasar";
+
+const crops: Ref<Crop[]> = storeToRefs(cropsStore()).getCrops;
+let input = ref("");
+
+const props = defineProps<{
+  title: string;
+  visibleColumns: string[] | undefined;
+}>();
```
[the remark this answers is not in the document]
Comment (fmarahre) [reply vue/src/components/CropsTable.vue:234]: It's from an older version. I forgot to delete it.
 - 'undefined' can be removed

File: vue/src/components/CropsTable.vue [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+import { cropsStore } from "../stores/cropsStore";
+import { ref } from "vue";
+import type { QTableProps } from "quasar";
+
+const crops: Ref<Crop[]> = storeToRefs(cropsStore()).getCrops;
+let input = ref("");
+
+const props = defineProps<{
+  title: string;
+  visibleColumns: string[] | undefined;
```
Comment (lucaslootalot) [inline vue/src/components/CropsTable.vue:233]: can be removed 
```suggestion
  visibleColumns: string[];
```

**Pull request merged**